backend/test_server.py

```
@@ -189,15 +189,297 @@ def make_post_request(endpoint, data):
+# ============================================================
+# Error Validation Tests
+# ============================================================
+
+def test_mcq_empty_input():
+    """Test rejection of empty input text."""
+    endpoint = '/get_mcq'
+    data = {
+        'input_text': '',
+        'max_questions': 5
+    }
+    response = requests.post(f'{BASE_URL}{endpoint}', headers={'Content-Type': 'application/json'}, data=json.dumps(data))
```

Ruff (0.15.7)

[error] Probable use of `requests` call without timeout

(S113)